ocaml/idl/datamodel_host_driver.ml

```diff
@@ -0,0 +1,86 @@
+(*
+   Copyright (c) Cloud Software Group, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+(** A Host_driver instance represents a driver on a host that is
+    installed in multiple versions. At most one version is active; a
+    different version can be selected to become active after reboot. If
+    no version is active, selecting a version makes it the active
+    version immediately. *)
+
+let select =
+  call ~name:"select" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "UNSUPPORTED. Select a variant of the driver to become active after \
+       reboot or immediately if currently no version is active"
+    ~params:
+      [
+        (Ref _host_driver, "self", "Driver to become active (after reboot).")
+      ; ( Ref _driver_variant
+        , "variant"
+        , "Driver version to become active (after reboot)."
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let deselect =
+  call ~name:"deselect" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "UNSUPPORTED. Deselect the currently active variant of this driver after \
+       reboot. No action will be taken if no variant is currently active."
```

Comment:
So selected and active are independent operations, as mentioned in the design. What happens when there is no active driver but there is a selected driver? Will it be deselected?

Reply:
A selected driver will become active on next reboot.

Reply:
I guess my confusion comes from this statement "Deselect the currently active variant of this driver", which says if the current driver is active, calling deselect on it will make it inactive? I was expecting something like "deselect the currently selected variant of this driver" because being selected and being active are two different states?

Reply:
Maybe the wording could be improved. What happens is that the currently active driver/variant combination will not become active again on reboot. Technically a symlink is removed such that this variant is no longer loaded by the kernel.